core/site_scons/tools.py

```diff
@@ -120,3 +121,48 @@ def get_defs_for_cmake(defs: list[str | tuple[str, str]]) -> list[str]:
         else:
             result.append(d)
     return result
+
+
+def _compress(data: bytes) -> bytes:
+    z = zlib.compressobj(level=9, wbits=-10)
+    return z.compress(data) + z.flush()
+
+
+def embed_binary(obj_program, env, section, target_, file):
+    _in = f"embedded_{section}.zip"
+    start_src = (
+        "_binary_build_firmware_" + _in.replace("/", "_").replace(".", "_") + "_start"
+    )
+    end_src = (
+        "_binary_build_firmware_" + _in.replace("/", "_").replace(".", "_") + "_end"
+    )
+    size_src = (
+        "_binary_build_firmware_" + _in.replace("/", "_").replace(".", "_") + "_size"
+    )
+    start_dest = "_binary_" + target_.replace("/", "_").replace(".o", "_zip") + "_start"
```

Comment:
'zip' file extension might be confusing as the file is not a real ZIP file. But this is really a nit.

Reply:
changed to 'bin', open to better suggestions https://github.com/trezor/trezor-firmware/commit/030e89abb801560a6223c0dcf6d5969ea6b6dacd

Reply:
I looked around but didn't find anything commonly used. I would like to propose using `.bin.deflated` to distinguish it from an uncompressed file.

Reply:
`.bin.deflated` seems good enough: https://github.com/trezor/trezor-firmware/commit/86ebbccd993ae23d00096089c15e698c87fb8e23